packages/core/src/dialects/abstract/query-generator-typescript.ts

```diff
@@ -779,32 +787,107 @@ export class AbstractQueryGeneratorTypeScript {
   }
 
   bulkDeleteQuery(tableName: TableNameOrModel, options: BulkDeleteQueryOptions): string {
-    const table = this.quoteTable(tableName);
-    const whereOptions = isModelStatic(tableName) ? { ...options, model: tableName } : options;
+    return joinSQLFragments([
+      `DELETE FROM ${this.quoteTable(tableName)}`,
+      this.#internals.addWhereSelectLimitOffset(tableName, options),
+    ]);
+  }
 
-    if (options.limit && this.dialect.supports.delete.modelWithLimit) {
-      if (!isModelStatic(tableName)) {
-        throw new Error('Cannot use LIMIT with bulkDeleteQuery without a model.');
+  /**
+   * Returns an update query
+   *
+   * @param tableName
+   * @param attrValueHash
+   * @param options
+   * @param columnDefinitions
+   */
+  updateQuery(
+    tableName: TableNameOrModel,
+    attrValueHash: Record<string, unknown>,
+    options?: UpdateQueryOptions,
+    columnDefinitions?: Record<string, AttributeOptions>,
```

Comment:
I would like to remove `columnDefinitions` from this method, as it feels too much like it was added after the fact on the old design:

Looking at what it does:

- It's used to map the attribute names specified in `attrValueHash` into column names. In our current design, if the user specifies a table identifier instead of a model or a model definition, we simply don't map. In migrations, the user needs to use column names instead of attribute names
- It's used to omit values of `autoIncremented` fields if specified. In my opinion, this is a mistake. Instead of removing it, it should throw an error explaining what happens, and it should only do so if a `model` or `modelDefinition` was provided, as no user is going to manually set the property `autoIncrement` to true in their update call
- It's used to produce `RETURNING`, but the user can already use the `returning` option themselves, and we can infer `returning` from the keys of `attrValueHash` if neither a `model`/`modelDefinition` nor a `returning` option were provided.
- It's used to serialize value based on the type. This is IMO the only good use, but that means we can drastically simplify this. Instead of having it as a fourth parameter, add it to `UpdateQueryOptions` as `columnTypes` which would be a `Record<string, DataType>`. This option can _only_ be provided if no model was provided

Reply:
I've changed to the columnTypes option however I reverted to only allow `DataTypeInstance` or `DataTypeClass` rather than allowing strings.

Reply:
Yes, good catch!